core/src/main/java/bisq/core/grpc/CoreOffersService.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * This file is part of Bisq.
+ *
+ * Bisq is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or (at
+ * your option) any later version.
+ *
+ * Bisq is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with Bisq. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package bisq.core.grpc;
+
+import bisq.core.monetary.Price;
+import bisq.core.offer.CreateOfferService;
+import bisq.core.offer.Offer;
+import bisq.core.offer.OfferBookService;
+import bisq.core.offer.OfferPayload;
+import bisq.core.offer.OpenOfferManager;
+import bisq.core.payment.PaymentAccount;
+import bisq.core.trade.handlers.TransactionResultHandler;
+import bisq.core.user.User;
+
+import org.bitcoinj.core.Coin;
+
+import javax.inject.Inject;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import lombok.extern.slf4j.Slf4j;
+
+import static bisq.core.offer.OfferPayload.Direction.BUY;
+
+@Slf4j
+public class CoreOffersService {
+
+    private final CreateOfferService createOfferService;
+    private final OfferBookService offerBookService;
+    private final OpenOfferManager openOfferManager;
+    private final User user;
+
+    @Inject
+    public CoreOffersService(CreateOfferService createOfferService,
+                             OfferBookService offerBookService,
+                             OpenOfferManager openOfferManager,
+                             User user) {
+        this.createOfferService = createOfferService;
+        this.offerBookService = offerBookService;
+        this.openOfferManager = openOfferManager;
+        this.user = user;
+    }
+
+    public List<Offer> getOffers(String direction, String fiatCurrencyCode) {
+        List<Offer> offers = offerBookService.getOffers().stream()
+                .filter(o -> !o.getDirection().name().equalsIgnoreCase(direction)
```

Comment:
There is an inconsistency here. If we call with `direction = sell`, this will return offers with `direction = buy`. Not yet sure how to fix it.

Reply:
Requested change is in commit 51d82b1.